python/pyarrow/io.pxi

```diff
@@ -44,6 +44,15 @@ cdef extern from "Python.h":
 
 
 cdef class NativeFile:
+    """
+    The base class for all Arrow streams.
+
+    Streams are either readable or writable.  They optionally support seeking.
```

Comment:
Though readability/writability aren't mutually exclusive (I guess the only implementation that supports both is MemoryMappedFile)

Reply:
Will fix :-)